catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/SearchController.java

```diff
@@ -176,15 +177,15 @@ public void executeQuery(
             try {
               long timeRemaining = Math.max(0, deadline - System.currentTimeMillis());
               future.get(timeRemaining, TimeUnit.MILLISECONDS);
-            } catch (InterruptedException e) {
-              LOGGER.info("Query interrupted for source {}", sourceId, e);
-              failSource(request, session, search, sourceId, e);
             } catch (TimeoutException e) {
               LOGGER.info("Query timed out for source {}", sourceId, e);
               failSource(request, session, search, sourceId, e);
             } catch (ExecutionException e) {
               LOGGER.info("Query failed for source {}", sourceId, e.getCause());
               failSource(request, session, search, sourceId, e);
+            } catch (InterruptedException e) {
+              LOGGER.info("Query interrupted for source {}", sourceId, e);
+              failSource(request, session, search, sourceId, e);
```

Comment:
❓  Should the thread's interrupted flag be set here as well?

Reply:
SonarQube doesn't have a problem with it so it isn't really part of my ticket. If I have time to look into it I'll definitely do that.